app/controllers/uploads_controller.rb

```diff
@@ -53,9 +45,30 @@ def show
 
   private
 
+  def alert_messages(loaded_csv)
+    total_rows_count = loaded_csv.ids.length
+    failed_rows = loaded_csv.failed_instances
+    failed_rows_count = failed_rows.length
+    valid_rows = total_rows_count - failed_rows_count
```

Comment:
Any chance these calculations could be moved out of the controller and into a Model perhaps?